src/backend/api/meals.js

```diff
@@ -1,15 +1,130 @@
 const express = require("express");
+const app = require("../app");
+//const { max } = require("../database");
 const router = express.Router();
 const knex = require("../database");
 
+//Get meals that has a price smaller than maxPrice
 router.get("/", async (request, response) => {
+  console.log("akjdskank");
+  if ("maxPrice" in request.query) {
+    const dataTable = await knex("meal").where(
+      "price",
+      "<",
+      Number(request.query.maxPrice)
+    );
+    return response.json(dataTable);
+  }
+
+  //Get meals that still has available reservations
+  if ("availableReservation" in request.query) {
+    const availableReservations = await knex("meal")
+      .join("reservation", "reservation.meal_id", "=", "meal.id")
+      .where("max_reservations", ">", "number_of_guests");
+    response.json(availableReservations);
+  }
+  // Get meals that partially match a title
+  if ("title" in request.query) {
+    const getTitle = await knex("meal").where("title", "like", "%pasta%");
```

Comment:
Great job but you might want to query by title and not a hardcoded 'pasta', What if I search for potato?
you'll need something like  ``const getTitle = await knex("meal").where("title", "like", `%${request.query.title}%`)`` .
You might also want to send some nice message to the user if there is no food that matches their query/title.